Sources/Valet/SecureEnclaveError.swift

```diff
@@ -0,0 +1,82 @@
+//  Created by Allison Moyer on 5/28/21.
+//  Copyright © 2021 Square, Inc. All rights reserved.
+//
+//  Licensed under the Apache License Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing software
+//  distributed under the License is distributed on an "AS IS" BASIS
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import LocalAuthentication
+
+public enum SecureEnclaveError: Int, CaseIterable, CustomStringConvertible, Error, Equatable {
+    /// Access to the secure enclave was not attempted because it was not available (due to biometrics not being available or enrolled
+    /// or being locked out, or the authentication context being invalidated)
+    case couldNotAccess
+
+    /// Access to the secure enclave was not attempted because the user failed to provide valid credentials.
+    case authenticationFailed
+
+    /// Access to the secure enclave was not attempted because the user canceled the prompt.
+    case userCancelled
+
+    /// Access to the secure enclave was not attempted because the user opted to the fallback option (i.e. 'Enter Password').
+    /// Callers should handle this as a custom alternative option to satisfying authentication.
+    case userFallback
+
+    /// Access to the secure enclave was not attempted because authentication was canceled by system
+    /// (e.g. another application went to foreground).
+    case systemCancel
```

Comment:
Thoughts on folding this error case into `userCancelled`? We [do that today in `KeychainError`](https://github.com/square/Valet/blob/a8bbee4645c2431b1226ddb0e2a937bcc8d5d7da/Sources/Valet/KeychainError.swift#L42) – `errSecAuthFailed` covers both authentication failure and also the app going into the background. See in-depth discussion about folding error messages in https://github.com/square/Valet/pull/152#discussion_r222064003.

Reply:
That sounds good to me